### PR TITLE
Update placeholder text to meet UX guidelines of W3C

### PR DESCRIPTION
**Pull request by jpaugh**
#4947 Change placeholder text from a description to an example value.

According to http://www.w3.org/html/wg/drafts/html/master/single-page.html#the-placeholder-attribute, the placeholder should not be used as a label; an example value

### CODE REVIEW

**Comment by DDaems** [comment]
Maybe change
`http://example.com/cat-photo`
to
`//example.com/cat-photo`

**Comment by ltegman** [comment]
@DDaems I feel like the protocol neutral format has the potential to confuse people, especially since it's not generally used by end users to enter URLs into forms, only on the back-end to code anchor links.

@jpaugh Is there any chance you could also add a label around the input element? I feel like we're not really gaining any accessibility benefit if all we did was make the placeholder stop serving the purpose of a label. Something like this seems to look okay in the tiny iPhone display:
```html
<label>Link: <input type="text" placeholder="http://example.com/cat-photo" required></label>
```

**Comment by jpaugh** [comment]
@ltegman I believe the accessibility benefit (in a production scenario) is that if label text is used as a placeholder, it might be misinterpreted by the accessibility software, and therefore misconstrued to the end-user.

IMO, the input would look much better with a label, but there are any number of ways to make the end result better, at the risk of overwhelming a beginner on the code side. Do you think the example would lose clarity without some kind of label?

tldr; stop here ;-)

To be clear, my purpose is not to make this tutorial code ultra-accessible or to follow the best UX principles; obviously, it doesn't try to do that. But, omitting details for sake of brevity is a bit different than getting details incorrect: it's extra important to be correct, since any tutorial is --- in a sense --- an application for a *person*, and each developer has a far greater impact than any single computer application.

It is worth mentioning that this would become the first way-point where labels are used; currently they are introduced in the radio button way-point, but are not explained there. (Which is fine, I guess.)

**Comment by ltegman** [comment]
@jpaugh I think you misunderstand the W3C's concern with using label text in a placeholder only. The concern is that the placeholder text is only visible until a user starts typing and may not be read by a screen-reader. If all you do is change the placeholder to not be the label we're actually degrading usability further by not ever labeling that input instead of temporarily labeling it.

I don't see a problem with adding a label to text input without introduction when we've already learned about them for radio buttons.

**Comment by BerkeleyTrue** [comment]
This PR is closed due to inactivity.

Please feel free to re-open when it is ready to be merged.

Happy Coding!

**Comment by RachelRVasquez** [comment]
Just saw this pull request and thought I could provide some more clarity on why placeholders alone can be concern and what alternate methods can be used with or without a placeholder.

There are a few reasons why placeholders raise some accessibility flags. Usually because often developers don't give them proper contrast. Another is because semantically, placeholders are meant to provide "hints" to users rather than be replacements for labels. Placeholders, as already mentioned, disappear on `:focus` and therefore cause issues due to users thinking the field might already be filled or those with memory impairments. Here's a [reference from the Paciello Group](https://www.paciellogroup.com/blog/2011/02/html5-accessibility-chops-the-placeholder-attribute/).

So to use a `placeholder` in an accessible way, there are [a few ways to go about it](https://www.aaron-gustafson.com/notebook/labeled-with-love/).

To answer @ltegman 's question, we can wrap a label around the `input` element. See this [in progress gist for a11y project](https://gist.github.com/RachelRVasquez/9aed0ddda6eaf675fd15) for reference as well as the [discussion around it](https://github.com/a11yproject/a11yproject.com/issues/170). Hope that helps!
